app/scripts/controllers/preferences.js

```diff
@@ -679,6 +681,25 @@ export default class PreferencesController {
     });
   }
 
+  _subscribeToInfuraAvailability() {
+    this.network.on(NETWORK_EVENTS.INFURA_IS_BLOCKED, () => {
+      this._setInfuraBlocked(true);
+    });
+    this.network.on(NETWORK_EVENTS.INFURA_IS_UNBLOCKED, () => {
+      this._setInfuraBlocked(false);
+    });
+  }
+
+  /**
+   * A setter for the `infuraBlocked` property
+   * @param {boolean} isBlocked - Bool indicating whether Infura is blocked
+   * @returns {Promise<boolean>} A promise of the updated blockage status
+   */
+  _setInfuraBlocked(isBlocked) {
+    this.store.updateState({ infuraBlocked: isBlocked });
```

Comment:
Nit: we should avoid a state update here if `infuraBlocked` is already set to the same value as `isBlocked`, to avoid refreshing the UI when there is no change

Reply:
done, will not be unnecessarily set now